src/mongocxx/pipeline.hpp

```diff
@@ -120,6 +120,16 @@ class MONGOCXX_API pipeline {
     pipeline& redact(bsoncxx::document::view_or_value restrictions);
 
     ///
+    /// Randomly selects the specified number of documents that pass into the stage and passes the
+    /// remaining documents to the next stage in the pipeline.
+    ///
+    /// @see http://docs.mongodb.org/manual/reference/operator/aggregation/sample/#pipe._S_sample
+    ///
+    /// @param set the number of input documents to select.
+    ///
+    pipeline& sample(std::int32_t size);
```

Comment:
The main reason is to stay consistent with the $skip operator. Plus, there is no point in debating about an unsigned integer since it will be encoded in BSON which do not support unsigned integers. Maybe we could change those int32_t for int64_t, but that should be done in one shot for all operators using them, in my opinion.